src/values.cpp

```diff
@@ -641,6 +645,27 @@ auto operator<<(std::ostream& os, const UnaryOrigin& self) -> std::ostream& {
     return os;
 }
 
+// struct NaryOrigin
+auto operator==(const NaryOrigin& lhs, const NaryOrigin& rhs) noexcept -> bool {
+    return lhs.values == rhs.values && lhs.location == rhs.location;
+}
+auto operator!=(const NaryOrigin& lhs, const NaryOrigin& rhs) noexcept -> bool {
+    return !(lhs == rhs);
+}
+auto operator<<(std::ostream& os, const NaryOrigin& self) -> std::ostream& {
+    os << "UnaryOrigin{ "
```

Comment:
```suggestion
    os << "NaryOrigin{ "
```

(or the new name we decide on)